src/components/Menu/MenuItem.tsx

```diff
@@ -0,0 +1,152 @@
+import { useRouter } from 'next/router';
+import { ReactElement, useCallback, useEffect, useState } from 'react';
+import { Link as AmplifyUILink, Flex, View } from '@aws-amplify/ui-react';
+import { IconExternalLink, IconChevron } from '@/components/Icons';
+import Link from 'next/link';
+import { Platform } from '@/data/platforms';
+
+enum Levels {
+  Category = 1,
+  Subcategory = 2,
+  Page = 3
+}
+
+const pathHelper = (path: string) => {
+  const prefix = 'pages/[platform]/';
+
+  const pathNoPrefix = path.slice(prefix.length).split('.')[0];
+
+  const lastIndex = pathNoPrefix.lastIndexOf('/index');
+
+  if (lastIndex > -1) {
+    return pathNoPrefix.substring(0, lastIndex);
+  } else {
+    return pathNoPrefix;
+  }
+};
```

Comment:
This is some logic to help massage the `route` property that is coming from the `PageNode` object in Directories. At the moment I'm using the file path of the `mdx` file and so I have to do some string manipulation to get the actual route.

I originally wanted to use the file path because it would be easier for a content author to just copy+paste that section

Talked to @jacoblogan about this though and he thinks it might be easier to just use the exact route. Does anyone have thoughts on this? You can see the `buildDirectory.mjs` for the `route` property

Reply:
Had a question about this too; I wonder if you can just use the route since the file path mimics the url structure now with our new IA?

Reply:
Yea, it's totally possible for me to use it. I can make that change for this (will update the directory object and remove this helper function)